TradeItIosTicketSDK2/TradeItYahooCryptoTradingTicketViewController.swift

```diff
@@ -418,55 +418,71 @@ class TradeItYahooCryptoTradingTicketViewController:
         case .orderAction:
             cell.detailTextLabel?.text = self.instrumentOrderCapabilities?.labelFor(field: .actions, value: self.order.action.rawValue)
         case .quantity:
+            guard let instrumentOrderCapabilities = self.instrumentOrderCapabilities else { return UITableViewCell() }
             let cell = cell as? TradeItNumericToggleInputCell
             let quantitySymbol = self.order.quantitySymbol
             cell?.configure(
                 onValueUpdated: { newValue in
                     self.order.quantity = newValue
-                    self.reload(row: .estimatedCost)
+                    self.reload(row: .estimatedChange)
                     self.setReviewButtonEnablement()
                 },
                 onQuantityTypeToggled: {
-                    let supportedOrderQuantityTypes = self.instrumentOrderCapabilities?.supportedOrderQuantityTypesFor(action: self.order.action)
+                    let supportedOrderQuantityTypes = instrumentOrderCapabilities.supportedOrderQuantityTypesFor(action: self.order.action)
 
-                    guard let supportedOrderQuantityTypeCount = supportedOrderQuantityTypes?.count,
-                        supportedOrderQuantityTypeCount > 0
+                    guard supportedOrderQuantityTypes.count > 0
                         else { return }
 
-                    let currentIndex = supportedOrderQuantityTypes?.index(of: self.order.quantityType ?? supportedOrderQuantityTypes?.first ?? .baseCurrency) as Int? ?? 0
-                    let nextIndex = (currentIndex + 1) % supportedOrderQuantityTypeCount
-                    let nextOrderQuantityType = supportedOrderQuantityTypes?[nextIndex] ?? supportedOrderQuantityTypes?.first ?? .baseCurrency
+                    let currentIndex = supportedOrderQuantityTypes.index(of: self.order.quantityType ?? supportedOrderQuantityTypes.first ?? .baseCurrency) as Int? ?? 0
+                    let nextIndex = (currentIndex + 1) % supportedOrderQuantityTypes.count
+                    let nextOrderQuantityType = supportedOrderQuantityTypes[safe: nextIndex] ?? supportedOrderQuantityTypes.first ?? .baseCurrency
```

Comment:
Is `[safe: ]` a swift thing or an Array extension?

Reply:
Array extension. Swift will throw an exception.